File: ThirdIteration/java/Warrior.java
import java.util.Optional;

public class Warrior extends Character implements Attribute {

    public Warrior(String name) {
        super(name);
        this.name = name;
    }
    @Override
    public void attack(Character character) {

        if (character.isAlive())
            character.health -= Helper.randomNumber(0, 9);

    }

    @Override
    public void heal(Character character) {


        if (character == this) {
            if (character.isAlive())
                character.health += 1;
        }

    }
}

File: RPG/src/Warrior.kt
import kotlin.random.Random

class Warrior(
    name: String,
    health: Int = 100,
) : Character(name, health) {

    fun attack(anEntity : Entity) =
        super.attack(anEntity, Random.nextInt(10))

    override fun isEnemy(aCharacter : Character) : Boolean {
        if (this == aCharacter)
            return true

        return super.isEnemy(aCharacter)
    }

    fun heal() {
        println("$this healed itself")
        if (0 < this.health && this.health < 100)
            this.health++
    }
}